src/test/java/unit/in/ac/bits/protocolanalyzer/persistence/repository/AnalysisRepositoryTest.java

```diff
@@ -0,0 +1,158 @@
+package unit.in.ac.bits.protocolanalyzer.persistence.repository;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Timer;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.data.elasticsearch.core.query.IndexQuery;
+
+import com.google.common.eventbus.EventBus;
+
+import in.ac.bits.protocolanalyzer.analyzer.event.BucketLimitEvent;
+import in.ac.bits.protocolanalyzer.persistence.repository.AnalysisRepository;
+import in.ac.bits.protocolanalyzer.persistence.repository.SaveRepository;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AnalysisRepositoryTest {
+	
+	@Mock
+	public SaveRepository saveRepo;
+
+	@Mock
+	public ExecutorService executorService;
+	
+	@Spy
+	public ArrayBlockingQueue<IndexQuery> queries = new ArrayBlockingQueue<IndexQuery>(10);
+	
+	@Spy
+	public Timer pullTimer;
+	
+	@Spy
+	public ArrayList<IndexQuery> currentBucket = new ArrayList<IndexQuery>(5);
+	
+	@Mock
+	public HashMap<String, String> envProperties;
+	
+	@InjectMocks
+	public AnalysisRepository analRepo;
+	
+	@Mock
+	public EventBus bus;
+	
+	@Mock
+	public IndexQuery query;
+	
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+	}
+	
+	@Test
+	public void analRepoAutoWiringCheck() {
+		assertThat(analRepo, notNullValue());
+		assertThat(analRepo.getSaveRepo(), notNullValue());
+		assertThat(analRepo.getExecutorService(), notNullValue());
+		assertThat(analRepo.getQueries(), notNullValue());
+		assertThat(analRepo.getPullTimer(), notNullValue());
+		assertThat(analRepo.getCurrentBucket(), notNullValue());
+		assertThat(analRepo.getEnvProperties(), notNullValue());
+	}
+	
+	@Test
+	public void analRepoConfigurationTest() {
+		doNothing().when(saveRepo).configure(bus);
```

Comment:
verification of this mock can be done between lines-87 and 88

Reply:
Which mock?

Reply:
saveRepo.configure()

Reply:
The saveRepo mock was verified in the autowiringTest written just above this one.